utils/src/test/java/software/amazon/awssdk/utils/async/FlatteningSubscriberTest.java

```diff
@@ -205,8 +207,9 @@ public void requestsFromDownstreamDoNothingAfterOnError() {
     @Test
     public void stochastic_dataFlushedBeforeOnComplete() {
         ExecutorService exec = Executors.newSingleThreadExecutor();
+        Instant end = Instant.now().plus(10, SECONDS);
         try {
-            for (int i = 0; i < 30_000_000; ++i) {
+            while (Instant.now().isBefore(end)) {
```

Comment:
🙌

Reply:
Is this reasonably long enough to catch random errors like 30m iterations would?

Reply:
It's less likely to catch errors. How much less depends on the hardware. On my mac, I don't recall how long it took to run these tests before, but if it was 2 minutes then we have a 25% chance of catching an error that we would have caught with the previous code. E.g. if the previous code had a 0.01% chance of catching an error, now it's a 0.0025% chance on my mac. That varies based on the hardware, of course, since we switched from iterations to time. Faster hardware will have a higher chance of catching the errors.

Tests that run when a developer is present are expensive, because we pay for both the developer and hardware to run them. We need to make sure these are short. In this case, this test will miss errors that occur with a very low likelihood. That was true before, but now it's even more true.

These tests are good to build a base level of confidence that things won't immediately explode If they go to production, but if we need even more assurance we should move them to be tests that run when a developer is not present. Canaries, for example. Then it's just a hardware cost plus the maintenance cost of the canaries.

I don't think that's necessary right now since we have higher-level canary tests that also exercise this code path and we don't have reason to suspect this is a problematic code path and it's not likely to be changed anytime soon. If we do think it'll be changed more often, then it might be worth creating a dedicated canary for it to raise our chance of catching the bug.